### ISSUE TITLE
[Feature] A collapsible widget like HTML5's details element

### ISSUE THREAD

**Issue by sharat87** [appsmithorg/appsmith#2888]

**Comment by hawisizu** [comment]
Description of what I would need: be able to have a form with sections that are collapsible, to not show all fields from scratch, but only title of sections at first. 

like this: 

![image](https://user-images.githubusercontent.com/48994518/107030165-8187d580-67b0-11eb-8bd7-a5bad2d37abd.png)

but with more fields in a section. I guess it is quite doable with tabs, so not super prio I think

**Comment by Nikhil-Nandagopal** [comment]
@hawisizu this is a great feature request. I think adding this as an enhancement to our List Widget makes sense here. Usually, the case with these types of widgets, the number of sections is variable and the information in each expanded state is similar with a few fields +/-. Does that seem right to you?

@jsartisan @riodeuno

**Comment by riodeuno** [comment]
@Nikhil-Nandagopal I agree that many use cases would work as an enhancement to list widget. However, I think a solution which enhances list widget would not be flexible enough. I would suggest a more generic approach where we have a "container-like" widget with a heading (collapse trigger) which can be toggled to show and hide children. These widgets can be used in the list widget's template to deal with the list use-case.

**Comment by Nikhil-Nandagopal** [comment]
@riodeuno suggesting this approach because a standalong widget will have trouble pushing the widgets below it on expand / contract. A list however can simply scroll.

**Comment by chromonav** [comment]
@Nikhil-Nandagopal  what if want to have a form inside an accordian

**Comment by Nikhil-Nandagopal** [comment]
@chromonav that should be possible. We'll allow you to add any widget inside this. Could you share an example of the UI you're trying to achieve? It'll help us build this effectively

**Comment by vishaldfreak** [comment]
@Nikhil-Nandagopal I'm trying to achieve something similar to have menu/submenu show in a side bar.
![](https://cdn.discordapp.com/attachments/1026638177454260264/1026673527098707998/Screen_Recording_2022-10-04_at_07.21.12_AM.gif)

**Comment by Nikhil-Nandagopal** [comment]
@vishaldfreak we're going to make this a default feature of our navigation where you can group pages to achieve this
